### PR TITLE
[GSoC] Joints framework documentation

### LINKED ISSUE #21705: Create a figure to explain the ``parent_axis``, ``child_axis``, ``parent_pos``, and ``child_pos`` vectors in ``Joint``.
The ``Joint`` class in mechanics has the ability to arbitrarily align and position the joint. For the non-trivial cases it is a bit difficult to understand. A figure is needed to help the user more clearly understand the relationship of the 4 vectors and the two connected bodies. I sketched this figure as a possible starting point:
P6220674

Ideally, we create an SVG figure that can be modified and stored in the codebase.

@sidhu1012, fyi

### PR DESCRIPTION
**Pull request by tjstienstra**
#### Brief description of what is fixed or changed
Fixes #21705.
Add general explanation about the joints framework within the physics/mechanics module.
Add images explaining the `PinJoint` and `PrismaticJoint`.
Add the `JointsMethod` docstring to the online documentation.

#### Release Notes

* other
  * Add documentation about the joints framework in the physics/mechanics module.

### CODE REVIEW

**Comment by TJStienstra** [comment]
@brocksam

**Comment by moorepants** [comment]
I'm seeing some issues in this image in the preview:
image

File: doc/src/modules/physics/mechanics/joints.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+===========================
+
+The general task of the :mod:`joints<sympy.physics.mechanics.joint>` is creating kinematic relationships between :class:`bodies<sympy.physics.mechanics.body.Body>`. Each joint has a setup as shown in the image below (this is the example of the :class:`~.PrismaticJoint`).
+
+.. image:: api/PrismaticJoint.*
+   :align: center
+   :width: 600
+
+As can be seen in this image, each joint needs several objects in order to define the relationships.
+First of it needs two bodies: the parent body (shown in green) and the child body (shown in blue).
```
Comment (moorepants) [inline doc/src/modules/physics/mechanics/joints.rst:19]:
```suggestion
First off it needs two bodies: the parent body (shown in green) and the child body (shown in blue).
```

File: doc/src/modules/physics/mechanics/joints.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+Both of these bodies have a masscenter from which the position of the joint is defined.
```
Comment (moorepants) [inline doc/src/modules/physics/mechanics/joints.rst:20]:
```suggestion
Both of these bodies have a mass center from which the position of the joint is defined.
```

File: doc/src/modules/physics/mechanics/joints.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+In the parent body the vector from the mass center to the ``parent_point`` is called the ``parent_joint_pos``.
+For the child body these are called the ``child_point`` and ``child_joint_pos``.
+The orientation of the joint in each body is defined by the ``parent_axis`` and ``child_axis``.
+These two vectors are aligned as explained in the :class:`~.Joint` notes and are in the image parallel to the orange vector.
```
Comment (moorepants) [inline doc/src/modules/physics/mechanics/joints.rst:24]: The color looks more red than orange to me.

File: doc/src/modules/physics/mechanics/joints.rst [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+As last the joint also needs :func:`~.dynamicsymbols` as generalized coordinates and speeds.
```
Comment (moorepants) [inline doc/src/modules/physics/mechanics/joints.rst:25]: I guess people could create these joints to just get static connections of bodies, so having a dynamicsymbol isn't strictly needed.

File: doc/src/modules/physics/mechanics/joints.rst [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+In the case of the :class:`~.PrismaticJoint` shown above, the generalized coordinate ``q_1`` distance along the joint axis.
+And the generalized speed ``u_1`` is its velocity.
+
+With the information listed above, the joint defines the following relationships.
+It first defines the kinematic differential equations, which relate the generalized coordinates to the generalized speeds.
+Next, it orients the parent and child body with respect to each other.
+After which it also defines their velocity relationships.
```
Comment (moorepants) [inline doc/src/modules/physics/mechanics/joints.rst:32]: Maybe note the position vectors are 1/2*P.x + 1/10*P.y, etc.

**Comment by moorepants** [comment]
Make both png and svg. Fix the few notes and this is good.

**Comment by moorepants** [comment]
I thought about this a bit more and may not regret merged the png file.
image

I'm not sure we have an other png's in the repo and I didn't check the size of this one. It is > 2 Mb.

**Comment by moorepants** [comment]
There are some more pngs, but this one is a bit excessive in size:
```
moorepants@nandi:sympy(master)$ find . -name *.png -exec ls -lh {} \;
-rw-rw-r-- 1 moorepants moorepants 8,7K mrt 30  2021 ./sympy/plotting/tests/test_region_or.png
-rw-rw-r-- 1 moorepants moorepants 7,8K mrt 30  2021 ./sympy/plotting/tests/test_region_not.png
-rw-rw-r-- 1 moorepants moorepants 9,8K mrt 30  2021 ./sympy/plotting/tests/test_region_xor.png
-rw-rw-r-- 1 moorepants moorepants 6,8K mrt 30  2021 ./sympy/plotting/tests/test_region_and.png
-rw-rw-r-- 1 moorepants moorepants 68K jul 15 09:39 ./doc/src/_static/sympylogo.png
-rw-rw-r-- 1 moorepants moorepants 164K mrt 30  2021 ./doc/src/_static/sympylogo_big.png
-rw-rw-r-- 1 moorepants moorepants 138K mrt 30  2021 ./doc/src/pics/consoleascii.png
-rw-rw-r-- 1 moorepants moorepants 119K mrt 30  2021 ./doc/src/pics/consoleunicode.png
-rw-rw-r-- 1 moorepants moorepants 37K mrt 30  2021 ./doc/src/pics/ipythonnotebook.png
-rw-rw-r-- 1 moorepants moorepants 239K mrt 30  2021 ./doc/src/pics/ipythonqtconsole.png
-rw-rw-r-- 1 moorepants moorepants 3,2K mrt 30  2021 ./doc/src/pics/pngview1.png
-rw-rw-r-- 1 moorepants moorepants 1,4M jul 15 09:40 ./doc/src/modules/physics/mechanics/api/PrismaticJoint.png
-rw-rw-r-- 1 moorepants moorepants 2,3M jul 15 09:40 ./doc/src/modules/physics/mechanics/api/PinJoint.png
-rw-rw-r-- 1 moorepants moorepants 68K jul 15 09:39 ./doc/src/modules/physics/mechanics/examples/multidof-holonomic.png
-rw-rw-r-- 1 moorepants moorepants 173K jul 15 09:40 ./doc/src/modules/physics/continuum_mechanics/truss_example.png
-rw-rw-r-- 1 moorepants moorepants 46K jul 15 09:39 ./doc/src/modules/physics/continuum_mechanics/ildshear.png
-rw-rw-r-- 1 moorepants moorepants 8,8K mrt 30  2021 ./doc/src/modules/physics/continuum_mechanics/allowed-sign-conventions.png
-rw-rw-r-- 1 moorepants moorepants 31K jul 15 09:39 ./doc/src/modules/physics/continuum_mechanics/ildreaction.png
```

**Comment by moorepants** [comment]
@TJStienstra  I know you have a nice way of drawing these figures, but I think we need to stick to svg figures that are editable. We already seem to have a not so great precedent set with some images in the physics docs.

**Comment by TJStienstra** [comment]
Okay, than I'll go and work on some SVGs this week to replace them.

**Pull request merged**